modules/declarative-configuration-examples.adoc

```diff
@@ -0,0 +1,129 @@
+    mode: auto <10>
+    clientID: CLIENT_ID
+    clientSecret: CLIENT_SECRET
+clientSecret: CLIENT_SECRET
+iap: <11>
+    audience: audience
+saml: <12>
+    spIssuer: sample.issuer.com
+    metadataURL: sample.provider.com/metadata
+saml: <13>
+    spIssuer: sample.issuer.com
+    cert: | <14>
+    ssoURL: saml.provider.com
+    idpIssuer: idp.issuer.com
+userpki:
+    certificateAuthorities: <15> |
+    certificate <16>
+----
+<1> Identifies the minimum role that will be assigned by default to any user logging in. If left blank, the value is `None`.
+<2> Use the user interface endpoint of your Central instance.
+<3> If your Central instance is exposed to different endpoints, specify them here.
+<4> These fields map users to specific roles, based on their attributes.
+<5> The key can be any claim returned from the authentication provider.
+<6> Identifies the role that the users are given. You can use a default role or a declaratively-created role.
+<7> Optional: Use these fields if attributes returned from the authentication provider are required; for example, if the audience is limited to a specific organization or group.
+<8> This section is required only for OpenID Connect (OIDC) authentication providers.
+<9> Identifies the expected issuer for the token.
+<10> Identifies the OIDC callback mode. Possible values are `auto`, `post`, `query`, and `fragment`. The preferred value is `auto`.
+<11> This section is required only for Google Identity-Aware Proxy (IAP) authentication providers.
+<12> This section is required only for Security Assertion Markup Language (SAML) 2.0 dynamic configuration authentication providers.
+<13> This section is required only for SAML 2.0 static configuration authentication providers.
+<14> Include the certificate in Privacy Enhanced Mail (PEM) format.
+<15> This section is required only for authentication with user certificates.
+<16> Include the certificate in PEM format.
+
+[id="declarative-config-example-permission-set"]
+== Declarative configuration permission set example
+
+.Declarative configuration permission set example
+
+[source,yaml]
+----
+name: A sample permission set
+description: A sample permission set created declaratively
+resources:
+- resource: Integration <1>
+  access: READ_ACCESS  <2>
+- resource: Administration
+  access: READ_WRITE_ACCESS
+----
+<1> For a full list of supported resources, navigate to *Access Control* -> *Permission Sets*.
+<2> Access can be either `READ_ACCESS` or `READ_WRITE_ACCESS`.
+
+[id="declarative-config-example-access-scope"]
+== Declarative configuration access scope example
+
+.Declarative configuration access scope example
+
+[source,yaml]
+----
+name: A sample access scope
+description: A sample access scope created declaratively
+rules:
+    included:
+        - cluster: secured-cluster-A <1>
+          namespaces:
+            - namespaceA
+        - cluster: secured-cluster-B
+    clusterLabelSelectors:
+        - requirements:
+            - key: a
```

Comment:
I think I provided a bad example in the doc input, sorry!

We can make it a bit more "obvious" by using something like this:
```yaml
key: kubernetes.io/metadata.name
operator: IN
values:
- production
- staging
- environment
```